qiskit/visualization/pulse_v2/plotters/matplotlib.py

```diff
@@ -0,0 +1,201 @@
+                                        '{val:.1f}'.format(val=chart.scale))
+                    self.ax.text(x=x[0], y=y[0], s=text, **data.styles)
+                elif isinstance(data, drawings.BoxData):
+                    xy = x[0], y[0]
+                    box = Rectangle(xy, width=x[1]-x[0], height=y[1]-y[0],
+                                    fill=True, **data.styles)
+                    self.ax.add_patch(box)
+                else:
+                    VisualizationError('Data {name} is not supported '
+                                       'by {plotter}'.format(name=data,
+                                                             plotter=self.__class__.__name__))
+            # axis break
+            for pos in axis_config.axis_break_pos:
+                self.ax.text(x=pos, y=current_y,
+                             s='//',
+                             ha='center',
+                             va='center',
+                             zorder=self.canvas.formatter['layer.axis_label'],
+                             fontsize=self.canvas.formatter['text_size.axis_break_symbol'],
+                             rotation=180)
+
+            # shift chart position
+            current_y += chart.vmin - margin_y
+
+        # remove the last margin
+        current_y += margin_y
+
+        y_max = self.canvas.formatter['margin.top']
+        y_min = current_y - self.canvas.formatter['margin.bottom']
+
+        # plot axis break line
+        for pos in axis_config.axis_break_pos:
+            self.ax.plot([pos, pos], [y_min, y_max],
+                         zorder=self.canvas.formatter['layer.fill_waveform'] + 1,
+                         linewidth=self.canvas.formatter['line_width.axis_break'],
+                         color=self.canvas.formatter['color.background'])
+
+        # label
+        self.ax.set_xticks(list(axis_config.axis_map.keys()))
+        self.ax.set_xticklabels(list(axis_config.axis_map.values()),
+                                fontsize=self.canvas.formatter['text_size.axis_label'])
+        self.ax.set_xlabel(axis_config.label,
+                           fontsize=self.canvas.formatter['text_size.axis_label'])
+
+        # boundary
+        self.ax.set_xlim(*axis_config.window)
+        self.ax.set_ylim(y_min, y_max)
+
+        # title
+        if self.canvas.fig_title:
+            self.ax.text(x=axis_config.window[0],
+                         y=y_max,
+                         s=self.canvas.fig_title,
+                         ha='left',
+                         va='bottom',
+                         zorder=self.canvas.formatter['layer.fig_title'],
+                         color=self.canvas.formatter['color.fig_title'],
+                         size=self.canvas.formatter['text_size.fig_title'])
+
+    def save_file(self, filename: str):
+        """Save image to file.
+
+        Args:
+            filename: File path to output image data.
+        """
+        plt.savefig(filename,
```

Comment:
This is based on the current global matplotlib figure, rather this Class should own a figure that it saves.

Reply:
right, can this savefig method be removed (offloaded to user) or I need to fix this and keep the method?

Reply:
I think its good to have this as a method, I just think it should be called with https://matplotlib.org/api/_as_gen/matplotlib.figure.Figure.html#matplotlib.figure.Figure.savefig